pkg/scheduler/plugins/proportion/capacity_policy/capacity_policy.go

```diff
@@ -83,3 +113,131 @@ func getRequiredQuota(tasksToAllocate []*pod_info.PodInfo) *podgroup_info.JobReq
 	}
 	return &quota
 }
+
+// getFirstPendingPod returns the first pod in a job that is in Pending status.
+// This is used to avoid duplicate quota checks for the same job.
+func getFirstPendingPod(job *podgroup_info.PodGroupInfo) *pod_info.PodInfo {
+	for _, pod := range job.PodInfos {
+		if pod.Status == pod_status.Pending {
+			return pod
+		}
+	}
+	return nil
+}
+
+// OnSessionOpen is called when a new scheduling session begins. It registers
+// the early quota checking function that prevents jobs from being considered
+// for scheduling if they would exceed their parent queues' quotas.
+func (cp *CapacityPolicy) OnSessionOpen(ssn *framework.Session) {
+	// Register early quota checks
+	ssn.AddPrePredicateFn(func(task *pod_info.PodInfo, job *podgroup_info.PodGroupInfo) error {
+		// Only check for the first pending pod to avoid duplicate checks
+		firstPending := getFirstPendingPod(job)
+		if firstPending == nil || task != firstPending {
+			return nil
+		}
+
+		// Check parent queue quotas
+		return cp.checkParentQueueQuotas(job, ssn)
+	})
+}
+
+// checkParentQueueQuotas verifies that a job's resource requirements don't
+// exceed quotas at any level in its queue hierarchy. This includes:
+// - GPU quota checks
+// - CPU quota checks
+// - Memory quota checks
+//
+// The function traverses up the queue hierarchy starting from the job's
+// immediate parent queue. If any quota would be exceeded, it returns an
+// error with a detailed message.
+//
+// Note: Preemptible jobs (PriorityTrainNumber) are allowed to exceed parent
+// queue quotas, while non-preemptible jobs must strictly adhere to quotas.
+func (cp *CapacityPolicy) checkParentQueueQuotas(job *podgroup_info.PodGroupInfo, ssn *framework.Session) error {
+	// Skip quota checks for preemptible jobs
+	if job.Priority == constants.PriorityTrainNumber {
+		log.InfraLogger.V(5).Infof("Job: <%v/%v> is preemptible, skipping parent queue quota checks", job.Namespace, job.Name)
+		return nil
+	}
+
+	// Get queue info for this job
+	queue, found := ssn.Queues[job.Queue]
+	if !found {
+		return nil // Can't check quota without queue info
+	}
+
+	// Only check parent queues, not the job's direct queue
+	currentQueueID := queue.ParentQueue
+
+	for currentQueueID != "" {
+		parentQueue, found := ssn.Queues[currentQueueID]
+		if !found {
+			break
+		}
+
+		// Calculate job's total resource requirements
+		jobResources := resource_info.EmptyResource()
+		for _, pod := range job.PodInfos {
+			if pod.Status == pod_status.Pending {
+				jobResources.AddResourceRequirements(pod.ResReq)
+			}
+		}
+
+		// Check GPU quota
+		if parentQueue.Resources.GPU.Quota > 0 && jobResources.GPUs() > float64(parentQueue.Resources.GPU.Quota) {
+			errorMsg := fmt.Sprintf(
+				"parent queue '%s' quota has reached the allowable limit of GPUs. "+
```

Comment:
A queue can go over the quota, but not the limit. I would change the whole thing to address the limit.